node/testing.go

```diff
@@ -27,3 +34,41 @@ func TestNode(t *testing.T, tp Type, opts ...Option) *Node {
 	require.NoError(t, err)
 	return nd
 }
+
+// WithFaultMaker is used to enable test cases with invalid blocks
+func WithFaultMaker(faultHeight int64) Option {
+	log.Info("assign fault maker to get incorrect block at height ", faultHeight)
+	return func(sets *settings) {
+		sets.opts = append(sets.opts, fx.Replace(func(ctx context.Context,
+			b *types.Block,
+			comm *types.Commit,
+			vals *types.ValidatorSet,
+			dag format.NodeAdder) (*header.ExtendedHeader, error) {
+			var dah da.DataAvailabilityHeader
+
+			namespacedShares, _ := b.Data.ComputeShares()
+
+			extended, err := ipld.AddShares(ctx, namespacedShares.RawShares(), dag)
+			if err != nil {
+				return nil, err
+			}
+
+			if b.Height == faultHeight {
+				t := testing.T{}
+				size := len(namespacedShares.RawShares()) * 2
+				extended = ipld.RandEDS(&t, size)
+				shares := ipld.ExtractEDS(extended)
+				copy(shares[0][consts.NamespaceSize:], shares[1][consts.NamespaceSize:])
+				dah = ipld.CreateDAH(ctx, shares, uint64(extended.Width()), dag)
+			}
+			dah = da.NewDataAvailabilityHeader(extended)
+			eh := &header.ExtendedHeader{
+				RawHeader:    b.Header,
+				DAH:          &dah,
+				Commit:       comm,
+				ValidatorSet: vals,
+			}
+			return eh, nil
+		}))
+	}
+}
```

Comment:
When I proposed an idea to make an option to set _custom_ Maker I meant setting a _custom_ Maker, not specifically a Faulty Maker.  To be honest, It feels strange to me to make an option specifically for one test case in the Swamp, even though it can be later enabled in the Testgound as well(cc @Bidon15).

Let's:
* Make an option `WithHeaderMaker(header.Maker)` which allows setting a custom header.Maker via fx.Replace. It should not be part of the testing.go in the node pkg, but a regular option.
* Extract the faultMaking logic into `header/testing.go` as `FraudMaker`, so that (1) you can import it in the Swamp test, @Bidon15 can import it in testgound tests and (2) its clear by the name that this maker causes FraudProof, not some arbitrary fault.